src/client/client.c

```diff
@@ -289,6 +294,171 @@ int method_lifecycle_action_on(Client *client, char *node_name, char *unit, char
         return r;
 }
 
+int method_metrics_toggle(Client *client, char *method) {
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        int r = 0;
+        char *result = NULL;
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        HIRTE_INTERFACE_BASE_NAME,
+                        HIRTE_OBJECT_PATH,
+                        MANAGER_INTERFACE,
+                        method,
+                        &error,
+                        &message,
+                        "");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        r = sd_bus_message_read(message, "s", &result);
+        if (r < 0) {
+                fprintf(stderr, "Failed to parse response message: %s\n", strerror(-r));
+                return r;
+        }
+        printf("%s\n", result);
+        return r;
+}
+
+static int match_start_unit_job_metrics_signal(
+                sd_bus_message *m, UNUSED void *userdata, UNUSED sd_bus_error *error) {
+        const char *job_path = NULL;
+        const uint64_t job_measured_time = 0;
+        const uint64_t unit_net_start_time = 0;
+        const int64_t hirte_time = 0;
+        const char *node_name = NULL;
+        int r = 0;
+
+        r = sd_bus_message_read(
+                        m, "ssttx", &node_name, &job_path, &job_measured_time, &unit_net_start_time, &hirte_time);
+        if (r < 0) {
+                fprintf(stderr, "Can't parse job result: %s", strerror(-r));
+                return r;
+        }
+
+        printf("[%s] Start job %s:\n\t"
```

Comment:
I think having the actual unit here would be great to have the actual unit name here as well. Sure, we already have the correlation via the job path, but having the unit name here simplifies things a bit.

Reply:
I had the same thought while I was working on the demo. Will add

Reply:
Done